starky/src/evaluation_frame.rs

```diff
@@ -0,0 +1,133 @@
+use plonky2::field::packed::PackedField;
+use plonky2::field::types::Field;
+
+/// A trait for viewing an evaluation frame of a STARK table.
+///
+/// It allows to access the current and next rows at a given step
+/// and can be used to implement constraint evaluation both natively
+/// and recursively.
+pub trait StarkEvaluationFrame<F, P>: Sized
+where
+    F: Field,
+    P: PackedField<Scalar = F>,
+{
+    /// The number of columns for the STARK table this evaluation frame views.
+    const COLUMNS: usize;
+    const PUBLIC_INPUTS: usize;
+
+    /// Returns the local values (i.e. current row) for this evaluation frame.
+    fn get_local_values(&self) -> &[P];
```

Comment:
```suggestion
    fn local_values(&self) -> &[P];
```

But I'm not sure if that shorter name won't lead to clashes?